cinema/views.py

```diff
@@ -101,8 +104,26 @@ def get_serializer_class(self):
         if self.action == "retrieve":
             return MovieDetailSerializer
 
+        if self.action == "upload_image":
+            return MovieImageSerializer
+
         return MovieSerializer
 
+    @action(
+        detail=True,
+        methods=["POST"],
+        permission_classes=(IsAdminUser,),
+        url_path="upload_image",
+    )
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = MovieListSerializer(movie, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Make a following change
```suggestion
        movie = self.get_object()
        serializer = self.get_serializer(movie, data=request.data)
        serializer.is_valid(raise_exception=True)
        serializer.save()
        return Response(serializer.data, status=status.HTTP_200_OK)
```